### ISSUE TITLE
[WCAG] Documentation and Accessibility improvements across multiple components

### ISSUE THREAD

**Issue by kenchase** [Canadian-Geospatial-Platform/GeoView#3514]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Current Behavior

- Update accessibility.md with description of high-level accessibility features (keyboard nav, screen reader ARIA features, etc)

- Add a document that provides explanations for WCAG SC passes/fails

- Various small A11Y improvements across multiple components
